cparser/Elab.ml

```diff
@@ -1623,9 +1668,10 @@ let elab_expr vararg loc env a =
             match unroll env b1.etyp, unroll env b2.etyp with
             | (TPtr(ty, a) | TArray(ty, _, a)), (TInt _ | TEnum _) -> ty
             | (TInt _ | TEnum _), (TPtr(ty, a) | TArray(ty, _, a)) -> ty
-            | _, _ -> error "type error in binary '+'" in
-          if not (pointer_arithmetic_ok env ty) then
-            err "illegal pointer arithmetic in binary '+'";
+            | _, _ -> error "invalid operands to binary expression (%a and %a)"
```

Comment:
Again we lose a piece of context, namely that the "binary expression" in question is the "+" operator.